### ISSUE TITLE
openjdk-8-jre/jdk broken on armv*l

### ISSUE THREAD

**Issue by theofficialgman** [RPi-Distro/repo#263]
using the openjdk-8-jre package installed from the raspbian repo on armhf, running a simple java -version fails

the openjdk-11 package which comes from the debian bullseye repo works just fine

```
/usr/lib/jvm/java-8-openjdk-armhf/bin/java -version

#

# A fatal error has been detected by the Java Runtime Environment:

#

#  SIGBUS (0x7) at pc=0xe63bc828, pid=29085, tid=0xe608a460

#

# JRE version:  (8.0_212-b01) (build )

# Java VM: OpenJDK Client VM (25.212-b01 mixed mode linux-aarch32 )

# Problematic frame:

# V  [libjvm.so+0x15e828]

#

# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

#

# An error report file with more information is saved as:

# /home/username/hs_err_pid29085.log

#

# If you would like to submit a bug report, please visit:

#   http://bugreport.java.com/bugreport/crash.jsp

#

Aborted (core dumped)

```

installed version of the package:
```
Package: openjdk-8-jre

Status: install ok installed

Priority: optional

Section: java

Installed-Size: 229

Maintainer: OpenJDK Team <openjdk@lists.launchpad.net>

Architecture: armhf

Multi-Arch: same

Source: openjdk-8

Version: 8u212-b01-1+rpi1

Replaces: openjdk-8-jre-headless (<< 8u20~b20-3)

Provides: java2-runtime, java5-runtime, java6-runtime, java7-runtime, java8-runtime

Depends: openjdk-8-jre-headless (= 8u212-b01-1+rpi1), libglib2.0-0 (>= 2.24), libxrandr2, libxinerama1, libgl1-mesa-glx | libgl1, libgtk-3-0, libatk-wrapper-java-jni (>= 0.33.3-9~), libasound2 (>= 1.0.16), libatomic1 (>= 4.8), libc6 (>= 2.28), libgif7 (>= 5.1), libjpeg62-turbo (>= 1.3.1), libpng16-16 (>= 1.6.2-1), libpulse0 (>= 0.99.1), libx11-6, libxext6, zlib1g (>= 1:1.1.4)

Recommends: fonts-dejavu-extra

Suggests: icedtea-8-plugin

Description: OpenJDK Java runtime, using Hotspot JIT

 Full Java runtime environment - needed for executing Java GUI and Webstart

 programs, using Hotspot JIT.

 .

 The packages are built using the IcedTea build support and patches

 from the IcedTea project.

Homepage: http://openjdk.java.net/

```

**Comment by theofficialgman** [comment]
```
#

# A fatal error has been detected by the Java Runtime Environment:

#

#  SIGBUS (0x7) at pc=0xe63bc828, pid=29085, tid=0xe608a460

#

# JRE version:  (8.0_212-b01) (build )

# Java VM: OpenJDK Client VM (25.212-b01 mixed mode linux-aarch32 )

# Problematic frame:

# V  [libjvm.so+0x15e828]

#

# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

#

# If you would like to submit a bug report, please visit:

#   http://bugreport.java.com/bugreport/crash.jsp

#



---------------  T H R E A D  ---------------



Current thread (0xe5f07400):  JavaThread "Unknown thread" [_thread_in_vm, id=29086, stack(0xe603b000,0xe608b000)]



siginfo: si_signo: 7 (SIGBUS), si_code: 1 (BUS_ADRALN), si_addr: 0xe5f4fe0d



Registers:

R0=0x0000000a

R1=0xd3958170

R2=0xe5f4fe0d

R3=0xd3958150

R4=0x00000009

R5=0xe5f07400

R6=0xe6089820

R7=0x000000e5

R8=0x0000000a

R9=0xe686c000

R10=0xe686c000

R11=0xe6089454

R12=0xe5f4fe15

R13=0xe6089350

R14=0xe5f4fe0c

R15=0xe63bc828



Top of Stack: (sp=0xe6089350)

0xe6089350:   0000006e 0000000d 00000078 0000000d

0xe6089360:   e6804a30 e5f50fb8 e6804a00 e7ebe87c

0xe6089370:   00000000 e6804a5c 00000000 e6089398

0xe6089380:   00000000 e608a948 00000007 e6089400

0xe6089390:   e608965c e686c000 e5f4fde0 e5f50fb1

0xe60893a0:   e5f4fe0d e5f40f90 01001d00 00000000

0xe60893b0:   00000007 e6089400 e60893e4 e66aa9a0

0xe60893c0:   00000000 0000000b e686c000 00000060 



Instructions: (pc=0xe63bc828)

0xe63bc808:   48 ff ff 1a b4 20 1b e5 b8 30 1b e5 02 30 43 e0

0xe63bc818:   08 00 53 e3 10 01 00 9a 34 30 96 e5 08 c0 82 e2

0xe63bc828:   d0 00 c2 e1 b4 c0 0b e5 04 c0 93 e5 04 31 83 e0

0xe63bc838:   04 c0 8c e0 30 2f bf e6 31 1f bf e6 06 00 a0 e3 



Stack: [0xe603b000,0xe608b000],  sp=0xe6089350,  free space=312k

Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)

V  [libjvm.so+0x15e828]

V  [libjvm.so+0x15efa8]

V  [libjvm.so+0x1652cc]

V  [libjvm.so+0x16c384]

V  [libjvm.so+0x501e0c]

V  [libjvm.so+0x502764]

V  [libjvm.so+0x503e40]

V  [libjvm.so+0x504080]

V  [libjvm.so+0x546f68]

V  [libjvm.so+0x547ed8]

V  [libjvm.so+0x2982a4]

V  [libjvm.so+0x53b5f8]

V  [libjvm.so+0x303f2c]  JNI_CreateJavaVM+0x9c

C  [libjli.so+0x2218]

C  [libpthread.so.0+0x7300]  start_thread+0xd0

C  [libc.so.6+0xdf208]





---------------  P R O C E S S  ---------------



Java Threads: ( => current thread )



Other Threads:



=>0xe5f07400 (exited) JavaThread "Unknown thread" [_thread_in_vm, id=29086, stack(0xe603b000,0xe608b000)]



VM state:not at safepoint (not fully initialized)



VM Mutex/Monitor currently owned by a thread: None



GC Heap History (0 events):

No events



Deoptimization events (0 events):

No events



Classes redefined (0 events):

No events



Internal exceptions (0 events):

No events



Events (10 events):

Event: 0.073 loading class java/nio/Buffer done

Event: 0.073 loading class java/lang/Boolean

Event: 0.073 loading class java/lang/Boolean done

Event: 0.073 loading class java/lang/Character

Event: 0.075 loading class java/lang/Character done

Event: 0.075 loading class java/lang/Float

Event: 0.075 loading class java/lang/Number

Event: 0.076 loading class java/lang/Number done

Event: 0.076 loading class java/lang/Float done

Event: 0.076 loading class java/lang/Double





Dynamic libraries:

00010000-00011000 r-xp 00000000 b3:02 6161630                            /usr/lib/jvm/java-8-openjdk-armhf/jre/bin/java

00020000-00021000 r--p 00000000 b3:02 6161630                            /usr/lib/jvm/java-8-openjdk-armhf/jre/bin/java

00021000-00022000 rw-p 00001000 b3:02 6161630                            /usr/lib/jvm/java-8-openjdk-armhf/jre/bin/java

00022000-00043000 rw-p 00000000 00:00 0                                  [heap]

d3665000-d36b3000 rw-p 00000000 00:00 0 

d36b3000-d3882000 r--s 03c3b000 b3:02 6161718                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/rt.jar

d3882000-d3b03000 rw-p 00000000 00:00 0 

d3b03000-d3d29000 ---p 00000000 00:00 0 

d3d29000-d3d2f000 rw-p 00000000 00:00 0 

d3d2f000-d3d7f000 ---p 00000000 00:00 0 

d3d7f000-d3d82000 rw-p 00000000 00:00 0 

d3d82000-d3da9000 ---p 00000000 00:00 0 

d3da9000-d3daf000 rw-p 00000000 00:00 0 

d3daf000-d3dff000 ---p 00000000 00:00 0 

d3dff000-d4350000 rw-p 00000000 00:00 0 

d4350000-d9350000 ---p 00000000 00:00 0 

d9350000-d9e00000 rw-p 00000000 00:00 0 

d9e00000-e3e00000 ---p 00000000 00:00 0 

e3e08000-e3e0b000 rw-p 00000000 00:00 0 

e3e0b000-e3e88000 ---p 00000000 00:00 0 

e3e88000-e3f40000 rwxp 00000000 00:00 0 

e3f40000-e5e88000 ---p 00000000 00:00 0 

e5e88000-e5e8f000 r-xp 00000000 b3:02 6161678                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libzip.so

e5e8f000-e5e9e000 ---p 00007000 b3:02 6161678                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libzip.so

e5e9e000-e5e9f000 r--p 00006000 b3:02 6161678                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libzip.so

e5e9f000-e5ea0000 rw-p 00007000 b3:02 6161678                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libzip.so

e5ea0000-e5eaa000 r-xp 00000000 b3:02 5656365                            /lib/arm-linux-gnueabihf/libnss_files-2.31.so

e5eaa000-e5eb9000 ---p 0000a000 b3:02 5656365                            /lib/arm-linux-gnueabihf/libnss_files-2.31.so

e5eb9000-e5eba000 r--p 00009000 b3:02 5656365                            /lib/arm-linux-gnueabihf/libnss_files-2.31.so

e5eba000-e5ebb000 rw-p 0000a000 b3:02 5656365                            /lib/arm-linux-gnueabihf/libnss_files-2.31.so

e5ebb000-e5ec1000 rw-p 00000000 00:00 0 

e5ecb000-e5eef000 r-xp 00000000 b3:02 6161660                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libjava.so

e5eef000-e5efe000 ---p 00024000 b3:02 6161660                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libjava.so

e5efe000-e5eff000 r--p 00023000 b3:02 6161660                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libjava.so

e5eff000-e5f00000 rw-p 00024000 b3:02 6161660                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libjava.so

e5f00000-e5f7e000 rw-p 00000000 00:00 0 

e5f7e000-e6000000 ---p 00000000 00:00 0 

e6006000-e6011000 r-xp 00000000 b3:02 6161677                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libverify.so

e6011000-e6020000 ---p 0000b000 b3:02 6161677                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libverify.so

e6020000-e6021000 r--p 0000a000 b3:02 6161677                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libverify.so

e6021000-e6022000 rw-p 0000b000 b3:02 6161677                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/libverify.so

e6022000-e6028000 r-xp 00000000 b3:02 5656370                            /lib/arm-linux-gnueabihf/librt-2.31.so

e6028000-e6038000 ---p 00006000 b3:02 5656370                            /lib/arm-linux-gnueabihf/librt-2.31.so

e6038000-e6039000 r--p 00006000 b3:02 5656370                            /lib/arm-linux-gnueabihf/librt-2.31.so

e6039000-e603a000 rw-p 00007000 b3:02 5656370                            /lib/arm-linux-gnueabihf/librt-2.31.so

e603a000-e603b000 ---p 00000000 00:00 0 

e603b000-e603e000 ---p 00000000 00:00 0 

e603e000-e608b000 rw-p 00000000 00:00 0 

e608b000-e60a3000 r-xp 00000000 b3:02 5644272                            /lib/arm-linux-gnueabihf/libgcc_s.so.1

e60a3000-e60b2000 ---p 00018000 b3:02 5644272                            /lib/arm-linux-gnueabihf/libgcc_s.so.1

e60b2000-e60b3000 r--p 00017000 b3:02 5644272                            /lib/arm-linux-gnueabihf/libgcc_s.so.1

e60b3000-e60b4000 rw-p 00018000 b3:02 5644272                            /lib/arm-linux-gnueabihf/libgcc_s.so.1

e60b4000-e6112000 r-xp 00000000 b3:02 5656360                            /lib/arm-linux-gnueabihf/libm-2.31.so

e6112000-e6121000 ---p 0005e000 b3:02 5656360                            /lib/arm-linux-gnueabihf/libm-2.31.so

e6121000-e6122000 r--p 0005d000 b3:02 5656360                            /lib/arm-linux-gnueabihf/libm-2.31.so

e6122000-e6123000 rw-p 0005e000 b3:02 5656360                            /lib/arm-linux-gnueabihf/libm-2.31.so

e6123000-e6245000 r-xp 00000000 b3:02 5644278                            /usr/lib/arm-linux-gnueabihf/libstdc++.so.6.0.28

e6245000-e6255000 ---p 00122000 b3:02 5644278                            /usr/lib/arm-linux-gnueabihf/libstdc++.so.6.0.28

e6255000-e625a000 r--p 00122000 b3:02 5644278                            /usr/lib/arm-linux-gnueabihf/libstdc++.so.6.0.28

e625a000-e625c000 rw-p 00127000 b3:02 5644278                            /usr/lib/arm-linux-gnueabihf/libstdc++.so.6.0.28

e625c000-e625e000 rw-p 00000000 00:00 0 

e625e000-e684c000 r-xp 00000000 b3:02 6161644                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/client/libjvm.so

e684c000-e685c000 ---p 005ee000 b3:02 6161644                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/client/libjvm.so

e685c000-e686c000 r--p 005ee000 b3:02 6161644                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/client/libjvm.so

e686c000-e6884000 rw-p 005fe000 b3:02 6161644                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/client/libjvm.so

e6884000-e7c9f000 rw-p 00000000 00:00 0 

e7c9f000-e7dde000 r-xp 00000000 b3:02 5656358                            /lib/arm-linux-gnueabihf/libc-2.31.so

e7dde000-e7ded000 ---p 0013f000 b3:02 5656358                            /lib/arm-linux-gnueabihf/libc-2.31.so

e7ded000-e7def000 r--p 0013e000 b3:02 5656358                            /lib/arm-linux-gnueabihf/libc-2.31.so

e7def000-e7df1000 rw-p 00140000 b3:02 5656358                            /lib/arm-linux-gnueabihf/libc-2.31.so

e7df1000-e7df3000 rw-p 00000000 00:00 0 

e7df3000-e7df6000 r-xp 00000000 b3:02 5656359                            /lib/arm-linux-gnueabihf/libdl-2.31.so

e7df6000-e7e05000 ---p 00003000 b3:02 5656359                            /lib/arm-linux-gnueabihf/libdl-2.31.so

e7e05000-e7e06000 r--p 00002000 b3:02 5656359                            /lib/arm-linux-gnueabihf/libdl-2.31.so

e7e06000-e7e07000 rw-p 00003000 b3:02 5656359                            /lib/arm-linux-gnueabihf/libdl-2.31.so

e7e07000-e7e15000 r-xp 00000000 b3:02 6161646                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/jli/libjli.so

e7e15000-e7e24000 ---p 0000e000 b3:02 6161646                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/jli/libjli.so

e7e24000-e7e25000 r--p 0000d000 b3:02 6161646                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/jli/libjli.so

e7e25000-e7e26000 rw-p 0000e000 b3:02 6161646                            /usr/lib/jvm/java-8-openjdk-armhf/jre/lib/arm/jli/libjli.so

e7e26000-e7e38000 r-xp 00000000 b3:02 5647247                            /lib/arm-linux-gnueabihf/libz.so.1.2.11

e7e38000-e7e47000 ---p 00012000 b3:02 5647247                            /lib/arm-linux-gnueabihf/libz.so.1.2.11

e7e47000-e7e48000 r--p 00011000 b3:02 5647247                            /lib/arm-linux-gnueabihf/libz.so.1.2.11

e7e48000-e7e49000 rw-p 00012000 b3:02 5647247                            /lib/arm-linux-gnueabihf/libz.so.1.2.11

e7e49000-e7e61000 r-xp 00000000 b3:02 5656368                            /lib/arm-linux-gnueabihf/libpthread-2.31.so

e7e61000-e7e71000 ---p 00018000 b3:02 5656368                            /lib/arm-linux-gnueabihf/libpthread-2.31.so

e7e71000-e7e72000 r--p 00018000 b3:02 5656368                            /lib/arm-linux-gnueabihf/libpthread-2.31.so

e7e72000-e7e73000 rw-p 00019000 b3:02 5656368                            /lib/arm-linux-gnueabihf/libpthread-2.31.so

e7e73000-e7e75000 rw-p 00000000 00:00 0 

e7e75000-e7e7a000 r-xp 00000000 b3:02 5655808                            /usr/lib/arm-linux-gnueabihf/libatomic.so.1.2.0

e7e7a000-e7e89000 ---p 00005000 b3:02 5655808                            /usr/lib/arm-linux-gnueabihf/libatomic.so.1.2.0

e7e89000-e7e8a000 r--p 00004000 b3:02 5655808                            /usr/lib/arm-linux-gnueabihf/libatomic.so.1.2.0

e7e8a000-e7e8b000 rw-p 00005000 b3:02 5655808                            /usr/lib/arm-linux-gnueabihf/libatomic.so.1.2.0

e7e8b000-e7e8c000 rw-p 00000000 00:00 0 

e7e8c000-e7eae000 r-xp 00000000 b3:02 5656354                            /lib/arm-linux-gnueabihf/ld-2.31.so

e7eb0000-e7eb2000 rw-p 00000000 00:00 0 

e7eb3000-e7ebb000 rw-s 00000000 b3:02 1180325                            /tmp/hsperfdata_garrett/29085

e7ebb000-e7ebc000 r--p 00000000 00:00 0 

e7ebc000-e7ebe000 rw-p 00000000 00:00 0 

e7ebe000-e7ebf000 r--p 00022000 b3:02 5656354                            /lib/arm-linux-gnueabihf/ld-2.31.so

e7ebf000-e7ec0000 rw-p 00023000 b3:02 5656354                            /lib/arm-linux-gnueabihf/ld-2.31.so

ffaec000-ffb0d000 rw-p 00000000 00:00 0                                  [stack]

ffff0000-ffff1000 r-xp 00000000 00:00 0                                  [vectors]



VM Arguments:

java_command: <unknown>

java_class_path (initial): .

Launcher Type: SUN_STANDARD



Environment Variables:

PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games

SHELL=/bin/bash

DISPLAY=:2



Signal Handlers:

SIGSEGV: [libjvm.so+0x57b11c], sa_mask[0]=11111111011111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO

SIGBUS: [libjvm.so+0x57b11c], sa_mask[0]=11111111011111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO

SIGFPE: [libjvm.so+0x453b70], sa_mask[0]=11111111011111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO

SIGPIPE: [libjvm.so+0x453b70], sa_mask[0]=11111111011111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO

SIGXFSZ: [libjvm.so+0x453b70], sa_mask[0]=11111111011111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO

SIGILL: [libjvm.so+0x453b70], sa_mask[0]=11111111011111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO

SIGUSR1: SIG_DFL, sa_mask[0]=00000000000000000000000000000000, sa_flags=none

SIGUSR2: [libjvm.so+0x45398c], sa_mask[0]=00000000000000000000000000000000, sa_flags=SA_RESTART|SA_SIGINFO

SIGHUP: SIG_DFL, sa_mask[0]=00000000000000000000000000000000, sa_flags=none

SIGINT: SIG_DFL, sa_mask[0]=00000000000000000000000000000000, sa_flags=none

SIGTERM: SIG_DFL, sa_mask[0]=00000000000000000000000000000000, sa_flags=none

SIGQUIT: SIG_DFL, sa_mask[0]=00000000000000000000000000000000, sa_flags=none





---------------  S Y S T E M  ---------------



OS:PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"

NAME="Debian GNU/Linux"

VERSION_ID="11"

VERSION="11 (bullseye)"

VERSION_CODENAME=bullseye

ID=debian

HOME_URL="https://www.debian.org/"

SUPPORT_URL="https://www.debian.org/support"

BUG_REPORT_URL="https://bugs.debian.org/"



uname:Linux 4.9.140+ #511 SMP PREEMPT Tue Jul 20 18:41:57 EEST 2021 aarch64

libc:glibc 2.31 NPTL 2.31 

rlimit: STACK 8192k, CORE 0k, NPROC 13999, NOFILE 1048576, AS infinity

load average:1.64 1.42 1.54



/proc/meminfo:

MemTotal:        4079184 kB

MemFree:          497564 kB

MemAvailable:    3027388 kB

Buffers:          245000 kB

Cached:          2248852 kB

SwapCached:        14712 kB

Active:          1936468 kB

Inactive:         960200 kB

Active(anon):      63000 kB

Inactive(anon):   429256 kB

Active(file):    1873468 kB

Inactive(file):   530944 kB

Unevictable:       40208 kB

Mlocked:              16 kB

SwapTotal:       1048572 kB

SwapFree:         934216 kB

Dirty:                 0 kB

Writeback:             0 kB

AnonPages:        430440 kB

Mapped:           140672 kB

Shmem:             49372 kB

Slab:             255824 kB

SReclaimable:     212068 kB

SUnreclaim:        43756 kB

KernelStack:        8976 kB

PageTables:        10884 kB

NFS_Unstable:          0 kB

Bounce:                0 kB

WritebackTmp:          0 kB

CommitLimit:     3088164 kB

Committed_AS:    3334996 kB

VmallocTotal:   263061440 kB

VmallocUsed:           0 kB

VmallocChunk:          0 kB

AnonHugePages:    182272 kB

ShmemHugePages:        0 kB

ShmemPmdMapped:        0 kB

NvMapMemFree:     101440 kB

NvMapMemUsed:     117708 kB

CmaTotal:         475136 kB

CmaFree:           86320 kB

HugePages_Total:       0

HugePages_Free:        0

HugePages_Rsvd:        0

HugePages_Surp:        0

Hugepagesize:       2048 kB





CPU:total 4 (initial active 4) 0x41:0x1:0xd07:1



/proc/cpuinfo:

processor	: 0

model name	: ARMv8 Processor rev 1 (v8l)

BogoMIPS	: 38.40

Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32

CPU implementer	: 0x41

CPU architecture: 8

CPU variant	: 0x1

CPU part	: 0xd07

CPU revision	: 1



processor	: 1

model name	: ARMv8 Processor rev 1 (v8l)

BogoMIPS	: 38.40

Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32

CPU implementer	: 0x41

CPU architecture: 8

CPU variant	: 0x1

CPU part	: 0xd07

CPU revision	: 1



processor	: 2

model name	: ARMv8 Processor rev 1 (v8l)

BogoMIPS	: 38.40

Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32

CPU implementer	: 0x41

CPU architecture: 8

CPU variant	: 0x1

CPU part	: 0xd07

CPU revision	: 1



processor	: 3

model name	: ARMv8 Processor rev 1 (v8l)

BogoMIPS	: 38.40

Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32

CPU implementer	: 0x41

CPU architecture: 8

CPU variant	: 0x1

CPU part	: 0xd07

CPU revision	: 1







Memory: 4k page, physical 4079184k(497564k free), swap 1048572k(934216k free)



vm_info: OpenJDK Client VM (25.212-b01) for linux-aarch32 JRE (1.8.0_212-8u212-b01-1+rpi1-b01), built on Mar 29 2019 13:54:02 by "root" with gcc 8.3.0



time: Thu Sep 23 03:37:36 2021

timezone: UTC

elapsed time: 0 seconds (0d 0h 0m 0s)

```

**Comment by XECDesign** [comment]
You seem to be running this with an aarch64 kernel? IIRC, Java's cpu detection is pretty bad and it gets confused when you do that. Does it work if you run a 32bit kernel?

**Comment by theofficialgman** [comment]
Unfortunately the system I am testing on I can’t easily run a 32bit kernel. I can say that this might have been fixed in upstream because I have used this binary package of java 8 on someones github that does work and seems to have a newer build version. https://github.com/gl91306/lunar/tree/master/jre

**Comment by XECDesign** [comment]
If you're aware of a patch that fixes the issue, I could pull it in, but as is, we don't currently do our own builds, nor does the package come from our repo. You could try reproducing the issue running Debian, then reporting it there.

As is, this is a wontfix.

**Comment by theofficialgman** [comment]
@XECDesign could you point out to me where the openjdk-8 builds are coming from on the raspbian repo? they aren't in the debian stable repos (debian stable doesn't have openjdk 8 on bullseye and buster) and I can't figure out exactly where the raspbian build comes from

to me it does seem that this is a custom build. this is the build number from raspbian 8u212-b01-1+rpi1and version 212 is NOT hosted anywhere else that I can find in other package repos. debian unstable repos host 302 and 252 on oldoldstable

**Comment by XECDesign** [comment]
They come from raspbian. but I believe they add all packages from the previous release into new ones. So the original source would've been debian + some patch from raspbian (indicated by the +rpi1 in the version number) and then it was left unmaintained.

**Comment by XECDesign** [comment]
Given that it's not even in Debian, we'll definitely not be patching this even if there's a fix.

**Comment by theofficialgman** [comment]
If its unmaintained, broken, and old don’t you think it should be removed from the repo?

**Comment by XECDesign** [comment]
Absolutely, but it's not in our repo.

https://bugs.launchpad.net/raspbian

**Comment by theofficialgman** [comment]
what do you mean its not in your repo? its right here: http://archive.raspbian.org/raspbian/pool/main/o/openjdk-8/
right in the archive.raspbian.org repo and its included in the bullseye and buster and probably older versions as well (haven't checked all the release files)

**Comment by XECDesign** [comment]
You're confusing Raspberry Pi OS and Raspbian. Raspbian is a community project that aims to rebuild Debian for armv6. Raspberry Pi OS, at least for armhf, is built on top of Raspbian, but they're completly separate projects.

This issue tracker is for Raspberry Pi OS. The repo for it is archive.raspberrypi.org, not archive.raspbian.org.

**Comment by theofficialgman** [comment]
oh my bad... I got this issue location from this page on raspbian https://www.raspbian.org/RaspbianBugs

I should file it at the launchpad. my bad and thanks for the help that you have given me

**Comment by XECDesign** [comment]
Best of luck
